nova/virt/vmwareapi/driver.py

```diff
@@ -363,6 +367,81 @@ def check_can_live_migrate_destination(self, context, instance,
         data = objects.VMwareLiveMigrateData()
         data.cluster_name = CONF.vmware.cluster_name
         data.datastore_regex = CONF.vmware.datastore_regex
+        data.host_ip = CONF.vmware.host_ip
+        data.host_username = CONF.vmware.host_username
+        data.host_password = CONF.vmware.host_password
+        data.instance_uuid = self._session.vim.service_content.about.instanceUuid
+
+        cert = ssl.get_server_certificate((CONF.vmware.host_ip, 443))
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+        data.thumbprint = x509.digest("sha1")
+
+        return data
+
+    def get_server_data(self, context, instance, migrate_data):
+
+        data = dict()
+        cluster_ref = vm_util.get_cluster_ref_by_name(self._session,
+                                                      CONF.vmware.cluster_name)
+        cluster_hosts = self._session._call_method(vutil,
+                                                   'get_object_property',
+                                                   cluster_ref, 'host')
+        cluster_datastores = self._session._call_method(vutil,
+                                                        'get_object_property',
+                                                        cluster_ref,
+                                                        'datastore')
+        LOG.debug("cluster_datastores: %s", cluster_datastores)
+        if not cluster_datastores:
+            LOG.warning('No datastores found in the destination cluster')
+            return None
+
+        ds_hosts = None
+        for ds in cluster_datastores.ManagedObjectReference:
+            ds_hosts = self._session._call_method(vutil, 'get_object_property',
+                                                  ds, 'host')
+
+            if ds_hosts:
+                break
+        data['datastore'] = cluster_datastores.ManagedObjectReference[0].value
+
+        for ds_host in ds_hosts.DatastoreHostMount:
+            for cluster_host in cluster_hosts.ManagedObjectReference:
+                if ds_host.key.value == cluster_host.value:
+                    data['host'] = cluster_host.value
+
+        networks = self._session._call_method(vutil,
+                                   'get_object_property',
+                                              vutil.get_moref(data['host'], "HostSystem"),
+                                   'network')
+
+        for network in networks[0]:
+            if network._type != 'Network':
+                net = self._session._call_method(vutil,
+                                           'get_object_property',
+                                            network,
+                                           'config')
+                for bridge_name in migrate_data.target_bridge_name:
+                    if net.name == bridge_name:
+                        data['portgroup_key'] = net.key
```

Comment:
Don't you need to collect here all the portgroup keys for each of the bridges?
Potentially, we can have one nic of a VM in one portgroup and another in a different one.